File: benchmark/ruby/lcs.rb
start = Time.now.to_f * 1000
s1 = "abcdefghij" * 10
s2 = "acegikmoqs" * 10
m, n = s1.length, s2.length
dp = Array.new(m + 1) { Array.new(n + 1, 0) }
(1..m).each do |i|
    (1..n).each do |j|
        if s1[i - 1] == s2[j - 1]
            dp[i][j] = dp[i - 1][j - 1] + 1
        else
            a = dp[i - 1][j]
            b = dp[i][j - 1]
            dp[i][j] = a > b ? a : b
        end
    end
end
finish = Time.now.to_f * 1000
puts "LCS: #{dp[m][n]}"
puts "Time: #{(finish - start).round}ms"

File: benchmark/php/lcs.php
<?php
$s1 = str_repeat("abcdefghij", 10);
$s2 = str_repeat("acegikmoqs", 10);
$start = hrtime(true);
$m = strlen($s1); $n = strlen($s2);
$dp = array_fill(0, $m + 1, array_fill(0, $n + 1, 0));
for ($i = 1; $i <= $m; $i++) {
    for ($j = 1; $j <= $n; $j++) {
        if ($s1[$i-1] == $s2[$j-1]) { $dp[$i][$j] = $dp[$i-1][$j-1] + 1; }
        else { $dp[$i][$j] = max($dp[$i-1][$j], $dp[$i][$j-1]); }
    }
}
$end = hrtime(true);
$ms = ($end - $start) / 1e6;
echo "LCS: " . $dp[$m][$n] . "
";
echo "Time: " . round($ms) . "ms
";
